### PR TITLE
Add support for on-demand only services

### PR DESCRIPTION
**Pull request by imobachgs**
Fixes [bsc#1104568](https://bugzilla.suse.com/show_bug.cgi?id=1104568).

### CODE REVIEW

File: library/systemd/src/lib/yast2/system_service.rb [hunk 1]
```diff
@@ -239,7 +239,8 @@ def currently_active?
     def start_modes
       @start_modes = [:on_boot, :manual, :on_demand] unless found?
       return @start_modes if @start_modes
-      @start_modes = [:on_boot, :manual]
+      @start_modes = [:manual]
+      @start_modes << :on_boot unless service.static?
```
Comment (jreidinger) [inline library/systemd/src/lib/yast2/system_service.rb]: does it make sense at all? If service is static, then it will always run like udev. so for me different should be on_boot and on_demand, but even for on_demand e.g. dbus will start automatic during system boot.

File: library/systemd/src/lib/yast2/systemd_unit.rb [hunk 2]
```diff
@@ -254,10 +254,10 @@ def initialize(systemd_unit, property_text)
...
         extract_properties
         self[:active?]     = ACTIVE_STATES.include?(active_state)
-        self[:running?]    = sub_state    == "running"
-        self[:loaded?]     = load_state   == "loaded"
-        self[:not_found?]  = load_state   == "not-found"
-        self[:static?]     = load_state   == "static"
+        self[:running?]    = sub_state  == "running"
+        self[:loaded?]     = load_state == "loaded"
+        self[:not_found?]  = load_state == "not-found"
+        self[:static?]     = unit_file_state == "static"
```
Comment (jreidinger) [inline library/systemd/src/lib/yast2/systemd_unit.rb]: why this change is needed?

File: library/systemd/test/yast2/system_service_test.rb [hunk 3]
```diff
@@ -302,6 +304,14 @@
       end
     end
 
+    context "when the service has no associated socket" do
```
Comment (jreidinger) [inline library/systemd/test/yast2/system_service_test.rb]: this looks like wrong description

File: library/systemd/src/lib/yast2/systemd_unit.rb (hunk 2 above)
Comment (imobachgs) [reply library/systemd/src/lib/yast2/systemd_unit.rb]: Because it was basically wrong.

File: library/systemd/src/lib/yast2/system_service.rb (hunk 1 above)
Comment (imobachgs) [reply library/systemd/src/lib/yast2/system_service.rb]: Think, for instance, about tftpd.Perhaps you do not want that service to be started.

Comment (jreidinger) [reply library/systemd/src/lib/yast2/system_service.rb]: well, documentation says that static services cannot be enabled neither disabled as it missing "Install" section. It means it can be run only as depency either for socket or by other services ( like udev ). So I am not sure if manual is right one, but neither on_boot sounds good for me. But for now we do not have anything better, so lets keep it with manual. But please ensure that we will never run enable/disable on it as it won't work.
